android/src/main/java/com/tailscale/ipn/ui/viewModel/PeerDetailsViewModel.kt

```diff
@@ -16,7 +16,13 @@ import com.tailscale.ipn.ui.util.TimeUtil
 data class PeerSettingInfo(val titleRes: Int, val value: ComposableStringFormatter)
 
 
-class PeerDetailsViewModel(val model: IpnModel, val nodeId: StableNodeID) : ViewModel() {
+class PeerDetailsViewModelFactory(private val nodeId: StableNodeID) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return PeerDetailsViewModel(nodeId) as T
```

Comment:
The OOM errors can be traced here (partially).  I was using a shared notifier and a shared netmap amongst all of the components and even then, I was running into OOM errors.  You're creating a new notifier for each view model which just compounds the issue.    Each netmap update on corp is like 3Mb of JSON to parse - and you're going to need to fetch that, parse it, just to extract the single node from it here.

We need to share the notifiers and the parsing amongst all the view models.  There is only one netmap and we should observe it in only one place.